docs/en/query_language/select.md

```diff
@@ -468,10 +468,42 @@ Don't mix these syntaxes.
 
 ClickHouse doesn't support the syntax with commas directly, so we don't recommend to use it. The algorithm tries to rewrite the query in terms of `CROSS` and `INNER` `JOIN` clauses and then proceeds the query processing. When rewriting the query, ClickHouse tries to optimize performance and memory consumption. By default, ClickHouse treats comma as an `INNER JOIN` clause and converts it to `CROSS JOIN` when the algorithm cannot guaranty that `INNER JOIN` returns required data.
 
-#### ANY or ALL Strictness
+#### Strictness
+
+- `ALL` — If the right table has several matching rows, the data will be multiplied by the number of these rows. This is the normal `JOIN` behavior for standard SQL.
+- `ANY` — If the right table has several matching rows, only the first one found is joined. If the right table has only one matching row, the results of queries with `ANY` and `ALL` keywords are the same.
+- `ASOF` — For uncertain join of time series by timestamp. Usage of `ASOF JOIN` is described below.
+
+**ASOF JOIN Usage**
+
+`ASOF JOIN` is useful when you need to join two events that lay near in time. For example, consider the following tables:
+
+```
+     table_1               table_2
+  event   | ev_time      event     | ev_time
+----------|-----------   ----------|-----------             
+         ...                      ...
+event_1_1 | 12:00        event_2_1 | 11:59
+         ...             event_2_2 | 12:30
+event_1_2 | 13:00        event_2_3 | 13:00
+         ...                      ...
+```
+
+`ASOF JOIN` takes the timestamp of an event from `table_1` and finds in `table_2` an event, which timestamp is closest (equal or less) to the timestamp of the event from `table_1`. In our example, `event_1_1` can be joined with the `event_2_1`, `event_1_2` can be joined with `event_2_3`, `event_2_2` cannot be joined.
+
+Tables for `ASOF JOIN` must have the timestamp column. The timestamp column cannot be alone. You can use `UInt32`, `UInt64`, and `DateTime` data types for this column.
```

Comment:
> must have the timestamp column

It's not truth. The last join key's type should be one of UInt32, UInt64, Float32, Float64, Date, DateTime for now.